src/main/java/ch/njol/skript/variables/FlatFileStorage.java

```diff
@@ -18,180 +18,259 @@
  */
 package ch.njol.skript.variables;
 
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.lang.Variable;
+import ch.njol.skript.log.SkriptLogger;
+import ch.njol.skript.registrations.Classes;
+import ch.njol.skript.util.ExceptionUtils;
+import ch.njol.skript.util.FileUtils;
+import ch.njol.skript.util.Task;
+import ch.njol.skript.util.Utils;
+import ch.njol.skript.util.Version;
+import ch.njol.util.NotifyingReference;
+import org.eclipse.jdt.annotation.Nullable;
+
 import java.io.BufferedReader;
 import java.io.File;
-import java.io.FileInputStream;
 import java.io.FileOutputStream;
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.io.OutputStreamWriter;
 import java.io.PrintWriter;
 import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
 import java.util.ArrayList;
 import java.util.Map.Entry;
 import java.util.TreeMap;
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 
-import org.eclipse.jdt.annotation.Nullable;
-
-import ch.njol.skript.Skript;
-import ch.njol.skript.config.SectionNode;
-import ch.njol.skript.lang.Variable;
-import ch.njol.skript.log.SkriptLogger;
-import ch.njol.skript.registrations.Classes;
-import ch.njol.skript.util.ExceptionUtils;
-import ch.njol.skript.util.FileUtils;
-import ch.njol.skript.util.Task;
-import ch.njol.skript.util.Utils;
-import ch.njol.skript.util.Version;
-import ch.njol.util.NotifyingReference;
-
 /**
- * TODO use a database (SQLite) instead and only load a limited amount of variables into RAM - e.g. 2 GB (configurable). If more variables are available they will be loaded when
- * accessed. (rem: print a warning when Skript starts)
- * rem: store null variables (in memory) to prevent looking up the same variables over and over again
- * 
- * @author Peter Güttinger
+ * A variable storage that stores its content in a
+ * comma-separated value file (CSV file).
+ */
+/*
+ * TODO use a database (SQLite) instead and only load a limited amount of variables into RAM - e.g. 2 GB (configurable).
+ *  If more variables are available they will be loaded when
+ *  accessed. (rem: print a warning when Skript starts)
+ *  rem: store null variables (in memory) to prevent looking up the same variables over and over again
  */
 public class FlatFileStorage extends VariablesStorage {
-	
-	@SuppressWarnings("null")
-	public final static Charset UTF_8 = Charset.forName("UTF-8");
-	
+
 	/**
-	 * A Lock on this object must be acquired after connectionLock (if that lock is used) (and thus also after {@link Variables#getReadLock()}).
+	 * The {@link Charset} used in the CSV storage file.
+	 */
+	public static final Charset FILE_CHARSET = StandardCharsets.UTF_8;
```

Comment:
Maybe just use `StandardCharsets.UTF_8` everywhere and then leave a note the file is UTF-8? It seems like a universal standard to use UTF-8.

Reply:
I do agree on UTF-8 being a standard, but I feel like having it as a constant field makes cleaner code, as otherwise the same constant needs to be written out specifically everywhere.